sdv/sequential/par.py

```diff
@@ -310,20 +311,26 @@ def _transform_sequence_index(self, data):
         sequence_index_context = sequence_index_context.rename(
             columns={self._sequence_index: f'{self._sequence_index}.context'}
         )
+
+        if _is_datetime_type(sequence_index[self._sequence_index]):
+            sequence_index[self._sequence_index] = cast_to_datetime64(
+                sequence_index[self._sequence_index]
+            ).astype(np.int64)
```

Comment:
Before pandas 2.2, datetime columns stored as object dtypes were silently converted to float dtype. To be consistent with the `fill_value` used in L330 or L332, the column has to be numerical.

Let me know if this makes sense